### PR TITLE
osd/PrimaryLogPG.cc: num_unfound must not be 0

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

File: src/osd/PrimaryLogPG.cc [hunk 1]
```diff
@@ -11155,7 +11155,7 @@ bool PrimaryLogPG::start_recovery_ops(
     info.last_complete = info.last_update;
   }
 
-  if (num_missing == num_unfound) {
+  if (num_unfound > 0 && num_missing == num_unfound) {
```
Comment (xiexingguo) [inline src/osd/PrimaryLogPG.cc]: This is not right. The original condition covers exact two cases:
(1) Primary does not missing anything;
(2) Primary does have some object missing, and all those objects are currently unfound.

In both cases we should recover replicas first.

So you might want to fix the comments below here instead?

Comment (shinobu-x) [reply src/osd/PrimaryLogPG.cc]: Thanks for your review.

> (2) Primary does have some object missing, and all those objects are currently unfound.

I might have missed some context.

**Comment by jdurgin** [comment]
The check is correct as-is, as @xiexingguo pointed out.

**Pull request closed without merging**